【WIP】我的书架功能

Inori-Lover opened the pull request:
close https://github.com/LightNovelShelf/Web/issues/17

wuyu8512 commented:
https://github.com/LightNovelShelf/Web/commit/fdb46601db70459dc99e213fa1df83df9d7bf695
简单写了俩，看看有没有需要别的

Inori-Lover commented:
> * 需要一个设置用来关闭开启进出文件夹的动画
> * 选中的圆点在不同平台的表现不一致
> * 在已经选中书籍情况下，无法右键移出
> * 列表的key，我看出不去什么问题，是否可以删掉？

1. 单独做一个开关意义不大，要做也是项目级别的统一开关；另外这个开关写起来有点麻烦，vue的过渡效果是通过模板来写的，要开关就代表slot模板要写两遍
2. todo
3. fixed
4. fixed
5. 圆点改坏了这个没看到，似乎没有提交；看git log没修改记录

wuyu8512 commented:
发现个bug，查看现有部署地址，直接进入书籍详情页面，发现 `加入书架` 按钮显示后又马上隐藏
如果在进入 `我的书架` 页面后，再进入书籍详情页面，则无此问题

Inori-Lover commented:
> 发现个bug，查看现有部署地址，直接进入书籍详情页面，发现 `加入书架` 按钮显示后又马上隐藏 如果在进入 `我的书架` 页面后，再进入书籍详情页面，则无此问题

@wuyu8512 也不算是个bug吧，db的读取是异步，更别说服务器读取；异步就代表一定有初始态露出来；
组件第一次挂载一定是等书架数据读取了才准确，第二次开始就没这个现象了

计划在写一个loading反馈，做了之后初始loading应该会相对好点

wuyu8512 commented:
> 

那看来来好方法是等服务器完成解析，由服务器来解析本书是否存在书架中

wuyu8512 commented:
差不多了，但还缺个东西，服务器只存文件夹结构和id，书的数据要重新请求的，差不多两种方向
- 我给前结构的时候带上数据，这种情况很难做分页，但我可以在拉取数据的时候同时把不存在的和用户无权限访问的书去掉
- 我只给前端结构，前端自己请求具体的内容，可以做分页，同时需要对等待请求的书籍做一个loading态，对请求后不存在的书籍需要进行删除
